src/workerd/api/crypto-impl-asymmetric.c++

```diff
@@ -1814,9 +1813,63 @@ public:
     return !!result;
   }
 
+  kj::Array<kj::byte> deriveBits(
+      SubtleCrypto::DeriveKeyAlgorithm&& algorithm,
+      kj::Maybe<uint32_t> resultBitLength) const override final {
+    JSG_REQUIRE(getAlgorithmName() == "X25519", DOMNotSupportedError, ""
+        "The deriveBits operation is not implemented for \"", getAlgorithmName(), "\".");
+
+    JSG_REQUIRE(getType() == "private"_kj, DOMInvalidAccessError, ""
+        "The deriveBits operation is only valid for a private key, not \"", getType(), "\".");
+
+    auto& publicKey = JSG_REQUIRE_NONNULL(algorithm.$public, TypeError,
+        "Missing field \"public\" in \"derivedKeyParams\".");
+
+    JSG_REQUIRE(publicKey->getType() == "public"_kj, DOMInvalidAccessError, ""
+        "The public key provided has type \"", publicKey->getType(), "\", not \"public\"");
```

Comment:
Nit: It would be a bit odd to say, "the public key provided has type "private", not public" ... 

```suggestion
        "The key provided has type \"", publicKey->getType(), "\", not \"public\"");
```